### ISSUE TITLE
inline_tests backend: “Error: Atom expected”

### ISSUE THREAD

**Issue by thierry-martinez** [ocaml/dune#1549]
I am trying to declare an inline test backend with dune as it is documented in
https://dune.readthedocs.io/en/latest/tests.html#example-of-backend

It works well when the backend is defined in the same package than the program that uses it, but now I would like to define the backend in its own package. So I tried to add a `public_name` field and an opam package definition to make `dune install` work. And it did, but when I try to use the backend in another package, I get the following error:

```
$ dune build
Info: creating file dune-project with this contents: (lang dune 1.5)
File "/Users/tmartine/.opam/4.07.1/lib/dune_inline_tests/dune_inline_tests.dune", line 5, characters 3-141:
5 |    ((runner_libraries ())
6 |     (flags :standard)
7 |     (generate_runner
8 |      (run sed "s/(\\*TEST:\\(.*\\)\\*)/let () = \\1;;/" %{impl-files}))))))
Error: Atom expected
Hint: dune files require less parentheses than jbuild files.
```
I put the files I wrote in the following repository to help to reproduce the error:
https://gitlab.inria.fr/tmartine/dune_inline_tests
The backend package is defined in the `backend/` directory, and an example package that leads to the error above is defined in the `example/` directory.

**Comment by diml** [comment]
Yh, there is indeed an issue with the generation of these files.

**Comment by diml** [comment]
I pushed a fix in #1551. Thanks for the reproduction case, I used it as a regression test.

**Comment by thierry-martinez** [comment]
Thank you for the quick fix! https://github.com/ocaml/dune/pull/1551 works for me indeed.
